swagger.yaml

```diff
@@ -62,209 +152,368 @@ paths:
+                  format: date
+                  example: "2018-09-20"
+                beginTime:
+                  description: The starting time of the meeting time for the course in 24 hour format.
+                  type: string
+                  example: "14:00:00"
+                endDate:
+                  description: The ending date of the meeting time schedule.
+                  type: string
+                  format: date
+                  example: "2018-11-30"
+                endTime:
+                  description: The starting time of the meeting time for the course in 24 hour format.
+                  type: string
+                  example: "14:50:00"
+                room:
+                  description: Room where the meeting time meets.
+                  type: string
+                  example: "100"
+                building:
+                  description: The building abbreviation of the meeting time.
+                  type: string
+                  example: "LINC"
+                buildingDescription:
+                  description: The name of the building.
+                  type: string
+                  example: "Learning Innovation Center"
+                campus:
+                  description: Campus location of the meeting time.
+                  type: string
+                  example: "Oregon State - Corvallis"
+                hoursPerWeek:
+                  description: The amount of hours this meeting time meets per week.
+                  type: number
+                  format: float
+                  example: 1.66
+                creditHourSession:
+                  description: Session credit hours for this meeting time.
+                  type: integer
+                  example: 2
+                meetsSunday:
+                  description: Whether this meeting time meets on Sunday.
+                  type: boolean
+                  example: false
+                meetsMonday:
+                  description: Whether this meeting time meets on Monday.
+                  type: boolean
+                  example: true
+                meetsTuesday:
+                  description: Whether this meeting time meets on Tuesday.
+                  type: boolean
+                  example: false
+                meetsWednesday:
+                  description: Whether this meeting time meets on Wednesday.
+                  type: boolean
+                  example: true
+                meetsThursday:
+                  description: Whether this meeting time meets on Thursday.
+                  type: boolean
+                  example: false
+                meetsFriday:
+                  description: Whether this meeting time meets on Friday.
+                  type: boolean
+                  example: true
+                meetsSaturday:
+                  description: Whether this meeting time meets on Saturday.
+                  type: boolean
+                  example: false
```

Comment:
No need to change, but what do you think if we return a `meetsDays` which is an array containing the days to meet? For example: `[1, 3, 5]` stands for meeting on Monday, Wednesday and Friday?

Reply:
Good suggestion. I thought about this for a bit actually. I tried to find a standard for representing a weekly schedule in a JSON API. All I could find is jCal, which doesn't really seem like it would fit for this use case. https://tools.ietf.org/html/rfc7265

I think I'll keep the various meetsSunday, meetsMonday, etc. fields as is. The array idea is good and definitely lighter weight, but I prefer doing it more explicitly here. For example, if someone looked at the API response without knowing the documentation, how would they know that 1=Monday, 2=Tuesday, etc.? Do weeks start at Sunday or Monday? Do weeks start at 0 or 1? As far as I can tell (and please let me know if I'm overlooking something), there isn't really a standard for representing the days of the week as integers.

Reply:
I was thinking to follow the way which the cron schedule uses to represent weekdays (0-6, 0 stands for Sunday, 1 for Monday, etc.), but yeah it doesn't mean is a standard at all. Actually, it is not necessary to use integers to represent weekdays, we can even use a string array `['Monday', 'Wednesday', 'Friday']` for it. In that way, it should be more explicitly and also lighter.

Reply:
@tsoliangwu0130 ahh, good idea. Going one step further, we could use the same abbreviations that the catalog uses: M, T, W, Th, F, Sa, Su

Reply:
sounds good 👍